plugin/hooks.go

```diff
@@ -13,29 +13,29 @@ import (
 // These assignments are part of the wire protocol used to trigger hook events in plugins.
 //
 // Feel free to add more, but do not change existing assignments. Follow the naming convention of
-// <HookName>Id as the autogenerated glue code depends on that.
+// <HookName>ID as the autogenerated glue code depends on that.
 const (
-	OnActivateId             = 0
-	OnDeactivateId           = 1
-	ServeHTTPId              = 2
-	OnConfigurationChangeId  = 3
-	ExecuteCommandId         = 4
-	MessageWillBePostedId    = 5
-	MessageWillBeUpdatedId   = 6
-	MessageHasBeenPostedId   = 7
-	MessageHasBeenUpdatedId  = 8
-	UserHasJoinedChannelId   = 9
-	UserHasLeftChannelId     = 10
-	UserHasJoinedTeamId      = 11
-	UserHasLeftTeamId        = 12
-	ChannelHasBeenCreatedId  = 13
-	FileWillBeUploadedId     = 14
-	UserWillLogInId          = 15
-	UserHasLoggedInId        = 16
-	UserHasBeenCreatedId     = 17
-	ReactionHasBeenAddedId   = 18
-	ReactionHasBeenRemovedId = 19
-	TotalHooksId             = iota
+	OnActivateID             = 0
```

Comment:
I think we should leave them as-is. We saw downstream breakages because certain consumers import the `plugin` package too.

Anything which is not `model` or `plugin` should be good to be changed.

Reply:
Interesting, I am sure the plugin package is imported but what would be the use of those ids from the outside? As far as I can see they are only used by the RPC layer. I am okay with being extra cautious of course but I've thought about this as well and couldn't really imagine a use of them.

Reply:
Gotcha. All good from my side if they aren't being used.

Quick FYI to @levb @hanzei on this. Since we broke it last time with the variable case change.

Reply:
I also _suppose_ that we can freely change them. @crspeller Could you please take a look given that you added this code?

Reply:
I don't think it will break existing plugins, since those IDs are only used within the server and the rpc client. The plugins don't use them directly, so they will all be updated at once. Cant guarantee that, but they shouldn't be. 
What will break however is exactly what the comment above this warns about. The generator of the RPC glue at `plugin/interface_generator/main.go` will break and needs to be modified to match.

Reply:
Thank you @crspeller. So we've essentially modified the generated file instead of the generator.

@haardikdharma10 , can you help changing the Id suffix in the generator file as well:

https://github.com/mattermost/mattermost-server/blob/8e71ab1ecb4d05df5c13652bd3f0c8c4e8d9b0e6/plugin/interface_generator/main.go#L294-L309

`{{.Name}}Id` becoming `{{.Name}}ID`

Then you can run `make pluginapi` to verify everything is generated properly.